buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/java/AvailableJavaInstallations.kt

```diff
@@ -66,71 +62,73 @@ private
 const val oracleJdk8 = "Oracle JDK 8"
 
 
-private
-const val oracleJdk7 = "Oracle JDK 7"
-
-
-open class AvailableJavaInstallations(project: Project, private val javaInstallationProbe: JavaInstallationProbe, private val jvmVersionDetector: JvmVersionDetector) {
+open class AvailableJavaInstallations(private val project: Project, private val javaInstallationProbe: JavaInstallationProbe, private val jvmVersionDetector: JvmVersionDetector) {
     private
     val logger = LoggerFactory.getLogger(AvailableJavaInstallations::class.java)
-    private
-    val javaInstallations: Map<JavaVersion, JavaInstallation>
 
     val currentJavaInstallation: JavaInstallation
     val javaInstallationForTest: JavaInstallation
+    val javaInstallationForCompilation: JavaInstallation
 
     init {
-        val resolvedJava7Home = resolveJavaHomePath(java7HomePropertyName, project)
-        val resolvedJava9Home = resolveJavaHomePath(java9HomePropertyName, project)
-        require(resolvedJava9Home != null || JavaVersion.current().isJava9Compatible) { "Building gradle on Java 7/8 requires $java9HomePropertyName system property or project property" }
-        val javaHomesForCompilation = listOfNotNull(resolvedJava7Home, resolvedJava9Home)
-        val javaHomeForTest = resolveJavaHomePath(testJavaHomePropertyName, project)
-        javaInstallations = findJavaInstallations(javaHomesForCompilation)
         currentJavaInstallation = JavaInstallation(true, Jvm.current(), JavaVersion.current(), javaInstallationProbe)
-        javaInstallationForTest = when (javaHomeForTest) {
-            null -> currentJavaInstallation
-            else -> detectJavaInstallation(javaHomeForTest)
-        }
+        javaInstallationForTest = determineJavaInstallation(testJavaHomePropertyName)
+        javaInstallationForCompilation = determineJavaInstallation(compileJavaHomePropertyName)
```

Comment:
We should use the current Java installation if it is Java 9 even if the java9Home property is set.